src/test/java/com/sharewalk/controller/UserControllerTest.java

```diff
@@ -0,0 +1,94 @@
+package com.sharewalk.controller;
+
+import com.sharewalk.model.User;
+import com.sharewalk.service.UserService;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class UserControllerTest {
+    @Rule
+    public MockitoRule mockito = MockitoJUnit.rule();
+
+    @Rule
+    public ExpectedException ex = ExpectedException.none();
+
+    @InjectMocks
+    private UserController instance;
+
+    @Mock
+    private UserService userService;
+
+    @Test
+    public void getUserWalksTest(){
+        //given
+        Long userId = Long.valueOf(1L);
+        User mockUser=mock(User.class);
+        List walks = new ArrayList<>();
+        when(userService.getUserById(userId)).thenReturn(mockUser);
+        when(userService.listUserWalks(userId)).thenReturn(walks);
+        //when
+        ResponseEntity responseEntity = instance.getUserWalks(userId, null);
+        //then
+        assertEquals(HttpStatus.OK, responseEntity.getStatusCode());
+        assertEquals(walks, (List)responseEntity.getBody());
+
+    }
+
+    @Test
+    public void getUserWalksStartsWithTest(){
+        //given
+        Long userId = Long.valueOf(1L);
+        User mockUser=mock(User.class);
+        List walks = new ArrayList<>();
+        when(userService.listUserWalks(userId,"walk 1" )).thenReturn(walks);
```

Comment:
Don't disregard IntelliJ's remarks - you can learn a lot from them. Mine is yellow because of unchecked assignment from raw list here ;)